File: C++/C++Basic/08.for-Loop-Exercise/07.trekking-mania.cpp
#include <iostream>

using namespace std;

int main()
{
    int climberGroupCount;
    cin >> climberGroupCount;

    int totalClimberCount = 0;
    int climbersMussalla = 0;
    int climbersMontBlanc = 0;
    int climbersKilimanjaro = 0;
    int climbersK2 = 0;
    int climbersEverest = 0;

    for (int i = 1; i <= climberGroupCount; i++)
    {
        int climberCount;
        cin >> climberCount;

        totalClimberCount += climberCount;

        if (climberCount <= 5)
        {
            climbersMussalla += climberCount;
        }
        else if (climberCount <= 12)
        {
            climbersMontBlanc += climberCount;
        }
        else if (climberCount <= 25)
        {
            climbersKilimanjaro += climberCount;
        }
        else if (climberCount <= 40)
        {
            climbersK2 += climberCount;
        }
        else if (climberCount >= 41)
        {
            climbersEverest += climberCount;
        }
    }

    double percentClimberMusalla = (climbersMussalla * 1.0 / totalClimberCount ) * 100.0;
    double percentClimberMontBlanc = (climbersMontBlanc * 1.0 / totalClimberCount) * 100.0;
    double percentClimberKilimanjaro = (climbersKilimanjaro * 1.0 / totalClimberCount) * 100.0;
    double percentClimberK2 = (climbersK2 * 1.0 / totalClimberCount) * 100.0;
    double percentClimberEverest = (climbersEverest * 1.0 / totalClimberCount) * 100.0;
    
    cout.setf(ios::fixed);
    cout.precision(2);

    cout << percentClimberMusalla << "%" << endl;
    cout << percentClimberMontBlanc << "%" << endl;
    cout << percentClimberKilimanjaro << "%" << endl;
    cout << percentClimberK2 << "%" << endl;
    cout << percentClimberEverest<< "%" << endl;

    return 0;
}

File: Python/Basic/08.For-Loop-Exersice/07.trekking_mania.py
climber_group_count = int(input())

climbers_mussalla = 0
climbers_montblanc = 0
climbers_kilimanjaro = 0
climbers_K2 = 0
climbers_everest = 0

total_climber = 0

for _ in range(climber_group_count):
    climber_count = int(input())

    total_climber += climber_count

    if climber_count <= 5:
        climbers_mussalla += climber_count
    elif climber_count <= 12:
        climbers_montblanc += climber_count
    elif climber_count <= 25:
        climbers_kilimanjaro += climber_count
    elif climber_count <= 40:
        climbers_K2 += climber_count
    else:
        climbers_everest += climber_count

percent_climber_musalla = climbers_mussalla / total_climber * 100
percent_climber_montblanc = climbers_montblanc / total_climber * 100
percent_climber_kilimanjaro = climbers_kilimanjaro / total_climber * 100
percent_climber_k2 = climbers_K2 / total_climber * 100
percent_climber_everest = climbers_everest / total_climber * 100

print(f"{percent_climber_musalla:.2f}%")
print(f"{percent_climber_montblanc:.2f}%")
print(f"{percent_climber_kilimanjaro:.2f}%")
print(f"{percent_climber_k2:.2f}%")
print(f"{percent_climber_everest:.2f}%")
